src/ui/save-html.js

```diff
@@ -27,155 +26,52 @@ const downloadLinks = [
-  // Move meta viewport, as it controls the rendering on mobile.
-  const metaViewport = cloneDoc.querySelector("meta[name='viewport']");
-  if (metaViewport && head.firstChild !== metaViewport) {
-    insertions.appendChild(metaViewport);
-  }
-  // Move charset to near top, as it needs to be in the first 512 bytes.
-  let metaCharset = cloneDoc.querySelector(
-    "meta[charset], meta[content*='charset=']"
-  );
-  if (!metaCharset) {
-    metaCharset = cloneDoc.createElement("meta");
-    metaCharset.setAttribute("charset", "utf-8");
-  }
-  insertions.appendChild(metaCharset);
-  // Add meta generator
-  const metaGenerator = cloneDoc.createElement("meta");
-  metaGenerator.name = "generator";
-  metaGenerator.content =
-    "ReSpec " + window.respecVersion || "Developer Channel";
-  insertions.appendChild(metaGenerator);
-  head.insertBefore(insertions, head.firstChild);
-  pub("beforesave", documentElement);
-}
-
-function cleanupHyper(node) {
-  // collect first, or walker will cease too early
-  const comments = [...walkTree(
-    node, NodeFilter.SHOW_COMMENT, 
-    comment => comment.textContent.startsWith("_hyper")
-  )];
-  for (const comment of comments) {
-    comment.parentNode.removeChild(comment);
-  }
-}
-
-function* walkTree(...args) {
-  const walker = document.createTreeWalker(...args);
-  while (walker.nextNode()) {
-    yield walker.currentNode;
-  }
-}
 
 function toDownloadLink(
   { id, href, fileName, title, type = "" } = { type: "" }
 ) {
-  const a = document.createElement("a");
-  a.classList.add("respec-save-button");
-  a.id = id;
-  a.href = href;
-  a.download = fileName;
-  a.type = type;
-  a.addEventListener("click", () => {
-    ui.closeModal();
-  });
-  const render = hyperHTML.bind(a);
-  return render`
-    ${title}
-  `;
+  return hyperHTML`
+    <a
+      href="${href}"
+      id="${id}"
+      download="${fileName}"
+      type="${type}"
+      class="respec-save-button"
+      onclick=${() => ui.closeModal()}
+    >${title}</a>`;
 }
 
-const div = document.createElement("div");
-div.classList.add("respec-save-buttons");
-const dialogRender = hyperHTML.bind(div);
 const saveDialog = {
   show(button) {
     document.respecIsReady.then(() => {
-      dialogRender`
-        ${downloadLinks.map(toDownloadLink)}
-      `;
+      const div = hyperHTML`
+        <div class="respec-save-buttons">
+          ${downloadLinks.map(toDownloadLink)}
+        </div>`;
       ui.freshModal(l10n[lang].save_snapshot, div, button);
     });
   },
 };
 
-export function exportDocument(format, mimeType) {
-  // Bug in babel won't export async function for some reason?
-  return document.respecIsReady.then(() => {
-    const dataURL = toDataURL(serialize(format), mimeType);
-    const encodedString = dataURL.replace(/^data:\w+\/\w+;charset=utf-8,/, "");
-    const decodedString = decodeURIComponent(encodedString);
-    return decodedString;
-  });
+const supportsDownload = "download" in document.createElement("a");
```

Comment:
I think we need to keep `export function exportDocument(...)`, even if it calls into the new exporter module, as I think “respec2html” relies on this one. We can console.warn this one is deprecated.

Reply:
Not sure why my comment ended up here, it should be up where `exportDocument()` is. In any case, make sure you also update `evaluateHTML()` in `tools/respecDocWriter.js`.

Reply:
can we import it from `exporter.js` and export it here? or move it here altogether? like:
``` js 
import { exportDocument } from "./exporter";
export exportDocument;
```

Saying this because I think `exportDocument` logically belongs to `exporter.js`

I didn't realize there were modules outside `src` folder that rely on these modules. Will check it and fix :slightly_smiling_face: